contracts/protocol/tokenization/APoRToken.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.6.12;
+
+import {AToken} from './AToken.sol';
+import {IAPoRToken} from '../../interfaces/IAPoRToken.sol';
+import {ILendingPool} from '../../interfaces/ILendingPool.sol';
+import {IChainlinkAggregatorV3} from '../../interfaces/IChainlinkAggregatorV3.sol';
+import {SafeMath} from '../../dependencies/openzeppelin/contracts/SafeMath.sol';
+import {IERC20Detailed} from '../../dependencies/openzeppelin/contracts/IERC20Detailed.sol';
+import {Errors} from '../libraries/helpers/Errors.sol';
+
+/**
+ * @title Aave's APoRToken (aToken with proof-of-reserves) Contract
+ * @notice AToken that checks reserves before minting
+ * @author Chainlink
+ */
+contract APoRToken is AToken, IAPoRToken {
+  using SafeMath for uint256;
+
+  uint256 public constant MAX_AGE = 7 days;
+  address public feed;
+  uint256 public heartbeat;
+
+  modifier onlyPoolAdmin {
+    require(
+      _msgSender() == ILendingPool(_pool).getAddressesProvider().getPoolAdmin(),
+      Errors.CALLER_NOT_POOL_ADMIN
+    );
+    _;
+  }
+
+  /**
+   * @notice Overriden mint function that checks the specified proof-of-reserves feed to
+   * ensure that the supply of the underlying assets is not greater than the reported
+   * reserves.
+   * @dev The proof-of-reserves check is bypassed if feed is not set.
+   * @param account The address to mint tokens to
+   * @param amount The amount of tokens to mint
+   */
+  function _mint(address account, uint256 amount) internal virtual override {
+    if (feed == address(0)) {
+      super._mint(account, amount);
+      return;
+    }
+
+    // Get latest proof-of-reserves from the feed
+    (, int256 answer, , uint256 updatedAt, ) = IChainlinkAggregatorV3(feed).latestRoundData();
+    require(answer > 0, Errors.AT_POR_INVALID_ANSWER);
+
+    // Check the answer is fresh enough (i.e., within the specified heartbeat)
+    uint256 heartbeat_ = heartbeat == 0 ? MAX_AGE : heartbeat;
```

Comment:
We made an optimization [here](https://github.com/smartcontractkit/bitcoin-token-smart-contracts/pull/1/files#diff-f39dd7bfb62e9b993a4b64a53196dfa679e7ede5b077d55c3e2d9fdce6cdccdaR68) to set the heartbeat at the default heartbeat if using 0 in the setter to make minting cheaper. You should be good to replicate that here.

Reply:
Done. NB: I added a constructor to set the heartbeat to the default `MAX_AGE`. This makes it so that `APoRToken` now runs a `constructor` in addition to the `AToken#initialize(...)` function. Not the greatest solution, but avoids having to copy the entire `AToken` implementation.